### ISSUE TITLE
.NET Core 3.0 and EF Core 3.0.0 support

### ISSUE THREAD

**Issue by FWorren** [SteffenMangold/EntityFrameworkCore.Cacheable#19]
The EF Core nuget packages needs to be updated to support .NET Core 3.0. When migrating from .NET Core 2.2 to 3.0 and EF Core to 3.0.0 assembly cannot be loaded:
```
System.TypeLoadException: Method 'get_Info' in type 'EntityFrameworkCore.Cacheable.CacheableOptionsExtension' from assembly 'EntityFrameworkCore.Cacheable, Version=2.0.1.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
```

**Comment by giabao-sys** [comment]
yes, I have same issue. I have to remove this reference to upgrade EF Core to 3.0.

**Comment by maxali** [comment]
This is still an issue for .NET Core 3.1

**Comment by WUKS87** [comment]
Still a problem with this, I'm looking in implementation.

**Comment by Tushar278** [comment]
when we will get fix ?
or any other workaround for these issue ?

**Comment by rochapablo** [comment]
Any update on this?

**Comment by mardemza** [comment]
Any update on this? I have same issue.

**Comment by darpanp** [comment]
Came across this when exploring Cache for EF Core 7.0.3. I am getting this error. Was this every resolved?

**Comment by basic-code-info** [comment]
core 2.2 to core 3.1, the same error. is there any update on this?

**Comment by Awraf** [comment]
I fixed that. I replaced IDbContextOptionsExtension from library to custom and then update DbContext registration to next

` services.AddDbContext<ApplicationContext>(options =>
            {
               ....
                options.ReplaceService<IQueryCompiler, CustomQueryCompiler>();
                ((IDbContextOptionsBuilderInfrastructure)options).AddOrUpdateExtension(new DbCacheableOptionsExtension(new MemoryCacheProvider()));
....
            });`

`public class DbCacheableOptionsExtension : IDbContextOptionsExtension
{
	ICacheProvider _cacheProvider;

	internal DbCacheableOptionsExtension(ICacheProvider cacheProvider)
	{
		_cacheProvider = cacheProvider;
	}

	public string LogFragment => $"Using {_cacheProvider.GetType().Name}";

	public bool ApplyServices(IServiceCollection services)
	{
		services.AddSingleton<ICacheProvider>(_cacheProvider);

		return false;
	}

	public long GetServiceProviderHashCode() => 0L;

	public void Validate(IDbContextOptions options)
	{
	}

	void IDbContextOptionsExtension.ApplyServices(IServiceCollection services)
	{
		throw new System.NotImplementedException();
	}

	/// <summary>
	///     The option set from the <see cref="DbContextOptionsBuilder.UseSecondLevelMemoryCache" /> method.
	/// </summary>
	public virtual ICacheProvider CacheProvider => _cacheProvider;

	public DbContextOptionsExtensionInfo Info => throw new System.NotImplementedException();}`
